kura/org.eclipse.kura.util/src/main/java/org/eclipse/kura/util/message/store/SqlMessageStoreHelper.java

```
@@ -0,0 +1,320 @@
+        this.utcCalendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+        this.connectionProvider = connectionProvider;
+        this.tableName = table;
+        this.queries = queries;
+        this.sanitizer = sanitizer;
+        this.sanitizedTableName = sanitizer.apply(tableName);
+    }
+
+    public void createTableAndIndexes() throws KuraStoreException {
+        execute(this.queries.getSqlCreateTable());
+        execute(this.queries.getSqlCreateIndex());
+
+        createIndex(this.sanitizer.apply(this.tableName + "_PUBLISHEDON"), this.sanitizedTableName,
+                "(PUBLISHEDON DESC)");
+        createIndex(this.sanitizer.apply(this.tableName + "_CONFIRMEDON"), this.sanitizedTableName,
+                "(CONFIRMEDON DESC)");
+        createIndex(this.sanitizer.apply(this.tableName + "_DROPPEDON"), this.sanitizedTableName, "(DROPPEDON DESC)");
+    }
+
+    public int getMessageCount() throws KuraStoreException {
+
+        return this.connectionProvider.withPreparedStatement(this.queries.getSqlMessageCount(), (c, stmt) -> {
+            try (final ResultSet rs = stmt.executeQuery()) {
+                if (rs.next()) {
+                    return rs.getInt(1);
+                } else {
+                    throw new SQLException("result set is empty");
+                }
+            }
+        }, "Cannot get message count");
+    }
+
+    protected void validate(String topic) throws KuraStoreException {
+        if (topic == null || topic.trim().length() == 0) {
+            throw new KuraStoreException(null, "topic must be not null and not empty");
+        }
+    }
+
+    public long store(String topic, byte[] payload, int qos, boolean retain, int priority)
+            throws KuraStoreException {
+        validate(topic);
+
+        final Timestamp now = new Timestamp(new Date().getTime());
+
+        return this.connectionProvider.withConnection(c -> {
+
+            long result = -1;
```

As for in one of the previous comments, should we throw an exception instead of returning -1?